controller/api/destination/server.go

```diff
@@ -455,17 +461,20 @@ func hasSuffix(slice []string, suffix []string) bool {
 	return true
 }
 
-func getPodOpaquePortsAnnotations(pod *corev1.Pod) (map[uint32]struct{}, error) {
+func getPodOpaquePortsAnnotations(pod *corev1.Pod) (map[uint32]struct{}, bool, error) {
```

Comment:
If this is just a nit I think I'm going to keep as is. As is, this function only tries to get the annotation from a pod. If it is absent, it don't think it should return a default set from something independent of the pod.

Also the fact that we only have two places where we need to check the result after makes it less of an issue. If we get up to three or four call sites, maybe it would make more sense.